magic/core/src/main/java/link/magic/android/core/relayer/WebViewWrapper.kt

```diff
@@ -147,9 +148,14 @@ class WebViewWrapper internal constructor(context: Context, private val urlBuild
                 resetContext()
             }
             InboundMessageType.MAGIC_MG_BOX_SEND_RECEIPT.toString() in response.msgType -> {
-                // When a receipt is received cancel previously invoked debounce call
+                // When a receipt is received, cancel previously invoked debounce call
                 debouncer.cancel()
             }
+            InboundMessageType.MAGIC_SEND_PRODUCT_ANNOUNCEMENT.toString() in response.msgType -> {
+                val TAG: String = "Magic SDK ${WebViewWrapper::class.java.name}"
+                val announcement = Gson().fromJson<AnnouncementMessage>(message, AnnouncementMessage::class.java)
```

Comment:
AnnouncementMessage can be simplified using ResponseData<Response<AnnouncementResult>>